rules/TypeDeclaration/Rector/ClassMethod/AddReturnArrayDocblockBasedOnArrayMapRector.php

```diff
@@ -0,0 +1,162 @@
+    {
+        return array_map(function ($item): int {
+            return $item->id;
+        }, $items);
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    /**
+     * @return int[]
+     */
+    public function getItems(array $items)
+    {
+        return array_map(function ($item): int {
+            return $item->id;
+        }, $items);
+    }
+}
+CODE_SAMPLE
+                )],
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class, Function_::class];
+    }
+
+    /**
+     * @param ClassMethod|Function_ $node
+     */
+    public function refactor(Node $node): null|Function_|ClassMethod
+    {
+        $returnsScoped = $this->betterNodeFinder->findReturnsScoped($node);
+
+        if ($this->hasNonArrayReturnType($node)) {
+            return null;
+        }
+
+        // nothing to return? skip it
+        if ($returnsScoped === []) {
+            return null;
+        }
+
+        // only returns with expr and no void
+        if (! $this->returnAnalyzer->hasOnlyReturnWithExpr($node, $returnsScoped)) {
+            return null;
+        }
+
+        $closureReturnTypes = [];
+
+        foreach ($returnsScoped as $returnScoped) {
+            if (! $returnScoped->expr instanceof FuncCall) {
+                return null;
+            }
+
+            $arrayMapClosure = $this->matchArrayMapClosure($returnScoped->expr);
+            if (! $arrayMapClosure instanceof Closure) {
+                return null;
+            }
+
+            if (! $arrayMapClosure->returnType instanceof Node) {
+                return null;
+            }
+
+            $closureReturnTypes[] = $this->staticTypeMapper->mapPhpParserNodePHPStanType($arrayMapClosure->returnType);
+        }
+
+        $returnType = $this->typeFactory->createMixedPassedOrUnionType($closureReturnTypes);
+        $arrayType = new ArrayType(new MixedType(), $returnType);
+
+        $functionLikePhpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
+
+        $hasChanged = $this->phpDocTypeChanger->changeReturnType($node, $functionLikePhpDocInfo, $arrayType);
+        if ($hasChanged) {
+            return null;
+        }
+
+        return $node;
+    }
+
+    private function hasNonArrayReturnType(ClassMethod|Function_ $functionLike): bool
+    {
+        if (! $functionLike->returnType instanceof Identifier) {
+            return false;
+        }
+
+        return $functionLike->returnType->toLowerString() !== 'array';
+    }
+
+    private function matchArrayMapClosure(FuncCall $funcCall): ?Closure
+    {
+        if (! $this->isName($funcCall, 'array_map')) {
+            return null;
+        }
+
+        if ($funcCall->isFirstClassCallable()) {
+            return null;
+        }
+
+        // lets infer strict array_map() type
+        $firstArg = $funcCall->getArgs()[0];
+        if (! $firstArg->value instanceof Closure) {
```

Comment:
this should allow `ArrowFunction` as well, eg:

```php
        return array_map(fn ($item): stdClass => $item, $items);
```

ref https://3v4l.org/35Uoq